### PR TITLE
feat(contact-center): emit AgentStateChange event

### PR DESCRIPTION
**Pull request by mkesavan13**
# COMPLETES [SPARK-584812](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-584812)

## This pull request addresses

Emitting Event for `AgentStateChange`

## by making the following changes

- Updating cc.ts to listen to socket message and emit only the `AgentStateChange` event with the same name
- Updating the sample app to receive this event and update the dropdown accordingly

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

- Tested by logging in as supervisor and changing to Available State and Meeting State
- Tested by logging in as supervisor and changing to Custom State created from the Admin portal

https://github.com/user-attachments/assets/0b2ca748-6569-4dd2-9d86-3ca973e2c1fe

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

## Summary by CodeRabbit

- **New Features**
	- Enhanced agent state management and wrap-up code functionality.
	- Introduced a dropdown for selecting wrap-up codes.
	- Added event listeners for agent state changes and incoming messages.

- **Bug Fixes**
	- Corrected comparison logic for setting auxiliary code IDs.

- **Tests**
	- Added tests for new event handling and wrap-up code dropdown functionality.

### CODE REVIEW

File: packages/@webex/plugin-cc/src/cc.ts [hunk 1]
```diff
@@ -284,6 +284,14 @@ export default class ContactCenter extends WebexPlugin implements IContactCenter
    */
   private setupEventListeners() {
     this.services.connectionService.on('connectionLost', this.handleConnectionLost.bind(this));
+    this.services.webSocketManager.on('message', (event) => {
```
Comment (Kesari3008) [inline packages/@webex/plugin-cc/src/cc.ts]: We should switch off this listener when station logout is done

File: docs/samples/contact-center/app.js [hunk 2]
```diff
@@ -205,7 +205,13 @@ function register() {
       taskEvents.detail.task = task;
       
       incomingCallListener.dispatchEvent(taskEvents);
-    })
+    });
+
+    webex.cc.on('AgentStateChange', (data) => {
```
Comment (Kesari3008) [inline docs/samples/contact-center/app.js]: Can we emit an event with a different name from our SDK, something like agent:stateChange

**Review by Kesari3008 — CHANGES_REQUESTED** [review]
Is it possible to add some screenshot or screenrecording for this to verify the testing done in the PR description?

**Comment by mkesavan13** [comment]
@coderabbitai review

**Review by Shreyas281299 — APPROVED** [review]
Changes look good, please address Priya comments

File: packages/@webex/plugin-cc/src/cc.ts (hunk 1 above)
Comment (mkesavan13) [reply packages/@webex/plugin-cc/src/cc.ts]: Done

File: docs/samples/contact-center/app.js (hunk 2 above)
Comment (mkesavan13) [reply docs/samples/contact-center/app.js]: Done

File: docs/samples/contact-center/app.js [hunk 3]
```diff
@@ -276,7 +276,13 @@ function register() {
...
 
       incomingCallListener.dispatchEvent(taskEvents);
-    })
+    });
+
+    webex.cc.on('AgentStateChange', (data) => {
+      if(data.type = 'AgentStateChangeSuccess') {
+        idleCodesDropdown.value = data.auxCodeId.trim() !== '' ? data.auxCodeId : 0;
+      }
+    });
```
[the remark this answers is not in the document]
Comment (mkesavan13) [reply docs/samples/contact-center/app.js]: Done

**Review by Kesari3008 — APPROVED** [review]

**Pull request merged**